### ISSUE TITLE
🚀 Feature: Use the Catalog presentation API

### ISSUE THREAD

**Issue by freben** [backstage/backstage#20955]
### 🔖 Feature description

Now that #17474 is merged, let's use the presentation APIs all over the place! This should help with ensuring that entities are presented uniformly with their titles / display names / etc, and with the same iconography everywhere if applicable.

### 🎤 Context

See #17474

### ✌️ Possible Implementation

Places in the code that use `EntityRefLink`/`EntityRefLinks` are already covered.

But there's still a fair number of places that just plain render `metadata.name`, or use `humanizeEntityRef`, or use some custom logic like that. Those should generally try to use the `EntityDisplayName` React component if possible, or use `EntityPresentation.promise` when in non-React code (which becomes available after #24595 is merged).

We'd appreciate the help with digging through the Backstage core and plugins, finding places where the presentation API should be used.

### 👀 Have you spent some time to check if this feature request has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

No, but I'm happy to collaborate on a PR with someone else

**Comment by hainenber** [comment]
Hi there, could I take up this one? Thanks!

**Comment by freben** [comment]
It's free for anyone to chip away at! Several small PRs is probably better than one large - easier to review. And it's probably going to take a few stabs before it's complete so may be worth not assigning to a single person perhaps 🤔

**Comment by hainenber** [comment]
I see, I was acquainted with other projects to ask for assignment firstly and not aware of contribution etiquette here

Let me take a stab at it. Thanks!

**Comment by freben** [comment]
No worries, glad you asked! A lot of times we do assign to avoid trampling on each others' feet, but this one may be diverse enough to try leaving it out to begin with. Let's try and then reevaluate if needed!

**Comment by handsamtw** [comment]
Hi @freben , I would love to help out this issue. After reviewing the codebase, I have a couple of questions.

Since humanizeEntityRef is a function returning a string, how is it suggested to replace EntityDisplayName, which is a React component?
I have lookup the place using EntityDisplayName, but I can't figure out how it related to the usage of humanizeEntityRef.

Thank you for the patience!

**Comment by freben** [comment]
Hi!

Yeah as I alluded to in the issue description "There are some cases where you aren't yet in a React tree"; those cases may need some rewriting where possible (moving work out of a function and instead into a child helper component or similar), or if you find that this isn't working out at all, we'd like to see examples of those cases to see if there's something we should do to extend or alter the API interface to cater to those cases.

**Comment by MihaiTabara** [comment]
@freben any major use cases left in here before we consider this done?

**Comment by RudrakshNanavaty** [comment]
Hello 👋 

I would love to contribute to this, if it is still being worked on.

I saw it has been stale since long, so if this isn't a priority for now, I'm also open to contribute to any other issue.

This would be the first time I'm contributing to this project, probably not the last.

**Comment by freben** [comment]
Hey - go right ahead, there are likely still a bunch of places where this could be needed.

**Comment by sri11223** [comment]
Hi! I'd like to contribute to this issue.

I have hands-on experience with Backstage Catalog from my internship at MetLife 
where I built DevX plugins and worked with the catalog system. I'm familiar with 
TypeScript and React, and would love to implement the Catalog presentation API.

Could you provide guidance on:
- Which specific components need the API integration?
- Are there existing examples I should reference?
- Expected timeline for review?

Thanks!

**Comment by himi-sh** [comment]
I would love to contribute to this if this feature is still needed.

**Comment by dyatko** [comment]
With #33576 and #33818, all `humanizeEntityRef` and `humanizeEntity` call sites in the core repo have been removed. Only the deprecated definitions, their tests, and re-exports in `@backstage/plugin-catalog-react` remain — ready to be cleaned up whenever the API surface is pruned.

The remaining work for this issue is migrating direct access to `metadata.title` and `spec.profile.displayName` in presentation contexts to use `EntityDisplayName`, `useEntityPresentation`, or `entityPresentationSnapshot` instead. There are roughly ~18 files in `plugins/` that reference each of these fields directly, though many are in backend collators, the presentation API internals, or test fixtures where direct access is appropriate.
